### PR TITLE
Added an optional output junction channel in STAR

### PR DESCRIPTION
**Pull request by praveenraj2018**
Added an optional output channel for getting chimeric junctions from STAR/align. This is required for the downstream analysis for fusion calling with STAR-fusion.

### CODE REVIEW

File: modules/star/align/meta.yml [hunk 1]
```diff
@@ -69,7 +69,12 @@ output:
       type: file
       description: STAR output tab file(s) (optional)
       pattern: "*.tab"
+  - tab:
```
Comment (drpatelh) [inline modules/star/align/meta.yml:72]:
```suggestion
  - junction:
```

File: tests/modules/star/align/main.nf [hunk 2]
```diff
@@ -5,7 +5,7 @@ nextflow.enable.dsl = 2
 include { STAR_GENOMEGENERATE              } from '../../../../modules/star/genomegenerate/main.nf' addParams( options: [args: '--genomeSAindexNbases 9'] )
 include { STAR_ALIGN                       } from '../../../../modules/star/align/main.nf'          addParams( options: [args: '--readFilesCommand zcat'] )
 include { STAR_ALIGN as STAR_FOR_ARRIBA    } from '../../../../modules/star/align/main.nf'          addParams( options: [args: '--readFilesCommand zcat --outSAMtype BAM Unsorted --outSAMunmapped Within --outBAMcompression 0 --outFilterMultimapNmax 50 --peOverlapNbasesMin 10 --alignSplicedMateMapLminOverLmate 0.5 --alignSJstitchMismatchNmax 5 -1 5 5 --chimSegmentMin 10 --chimOutType WithinBAM HardClip --chimJunctionOverhangMin 10 --chimScoreDropMax 30 --chimScoreJunctionNonGTAG 0 --chimScoreSeparation 1 --chimSegmentReadGapMax 3 --chimMultimapNmax 50'] )
```
Comment (drpatelh) [inline tests/modules/star/align/main.nf:7]:
```suggestion
include { STAR_GENOMEGENERATE               } from '../../../../modules/star/genomegenerate/main.nf' addParams( options: [args: '--genomeSAindexNbases 9'] )
include { STAR_ALIGN                        } from '../../../../modules/star/align/main.nf'          addParams( options: [args: '--readFilesCommand zcat'] )
include { STAR_ALIGN as STAR_FOR_ARRIBA     } from '../../../../modules/star/align/main.nf'          addParams( options: [args: '--readFilesCommand zcat --outSAMtype BAM Unsorted --outSAMunmapped Within --outBAMcompression 0 --outFilterMultimapNmax 50 --peOverlapNbasesMin 10 --alignSplicedMateMapLminOverLmate 0.5 --alignSJstitchMismatchNmax 5 -1 5 5 --chimSegmentMin 10 --chimOutType WithinBAM HardClip --chimJunctionOverhangMin 10 --chimScoreDropMax 30 --chimScoreJunctionNonGTAG 0 --chimScoreSeparation 1 --chimSegmentReadGapMax 3 --chimMultimapNmax 50'] )
```

**Review by drpatelh — APPROVED** [review]
Nice! Thanks! Feel free to merge when the tests pass.

**Comment by praveenraj2018** [comment]
> Nice! Thanks! Feel free to merge when the tests pass.

Thanks @drpatelh

**Pull request merged**